storage/doc.go

```diff
@@ -224,14 +224,34 @@ BucketHandle.SignedURL method.
 	fmt.Println(url)
 
 You can also sign a URL wihout creating a client. See the documentation of
-SignedURL for details.
+[SignedURL] for details.
 
 	url, err := storage.SignedURL(bucketName, "shared-object", opts)
 	if err != nil {
 	    // TODO: Handle error.
 	}
 	fmt.Println(url)
 
+# Credential detection for [BucketHandle.SignedURL] and [BucketHandle.GenerateSignedPostPolicyV4]
+
+If the GoogleAccessID and PrivateKey option fields are not provided, they will
+be automatically detected if any of the following are true:
+  - you are authenticated to the Storage Client with a service account's
+    downloaded private key, either directly in code or by setting the
+    GOOGLE_APPLICATION_CREDENTIALS environment variable (see [Other Environments]),
+  - your application is running on Google Compute Engine (GCE), or
+  - you are logged into [gcloud using application default credentials]
+    with [impersonation enabled].
+
+Detecting GoogleAccessID may not be possible if you are authenticated using a
+token source or using [option.WithHTTPClient]. In this case, you can provide a
+service account email for GoogleAccessID and the client will attempt to sign
+the URL or Post Policy using that service account.
+
+To sign without a custom signing function you require:
```

Comment:
I don't think "without a custom signing function" is really relevant here, the signBlob permissions at least are required regardless of the detection piece of things

Maybe:

```
To generate the signature, you must have:
- iam.serviceAccounts.signBlob permissions on the GoogleAccessID service account.
- the [IAM Service Account Credentials API] enabled (unless authenticating with a downloaded private key).
```